.github/workflows/workflow.yml

```diff
@@ -85,6 +85,9 @@ jobs:
       # Ensure Dune can build itself
       - run: opam exec -- ./dune.exe build -p dune --profile dune-bootstrap
 
+      # To get more recent versions of Coq we add this opam repo
+      - run: opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
```

Comment:
This can be expressed as `opam-repositories:` under `setup-ocaml` above (not sure about the exact syntax). This might be necessary for the dune cache to work.